archive/scripts/analysis_0_single_run/nbconverted/3_visualize_data.py

```diff
@@ -0,0 +1,517 @@
+
+
+# In[14]:
+
+
+# Convert 'num_experiments' into categories to preserve the ordering
+lst_num_experiments_str = [str(i) for i in lst_num_experiments]
+num_experiments_cat = pd.Categorical(all_data_df['num_experiments'], categories=['1', 'multiple'])
+
+# Convert 'comparison' into categories to preserve the ordering
+comparison_cat = pd.Categorical(all_data_df['comparison'], categories=lst_num_experiments_str)
+
+# Assign to a new column in the df
+all_data_df = all_data_df.assign(num_experiments_cat = num_experiments_cat)
+all_data_df = all_data_df.assign(comparison_cat = comparison_cat)
+
+
+# In[15]:
+
+
+all_data_df.columns = ['PC1', 'PC2', 'num_experiments', 'comparison', 'No. of experiments', 'Comparison']
+
+
+# In[16]:
+
+
+# Plot all comparisons in one figure
+g_pca = ggplot(all_data_df, aes(x='PC1', y='PC2')) + geom_point(aes(color='No. of experiments'), alpha=0.1) + facet_wrap('~Comparison') + labs(x = "PC 1", y = "PC 2", title = "PCA of experiment 1 vs multiple experiments") + theme_bw() + theme(
+    legend_title_align = "center",
+    plot_background=element_rect(fill='white'),
+    legend_key=element_rect(fill='white', colour='white'), 
+    plot_title=element_text(weight='bold')
+) \
++ guides(colour=guide_legend(override_aes={'alpha': 1})) \
++ scale_colour_manual(['#bdbdbd', '#b3e5fc'])
+
+print(g_pca)
+ggsave(plot = g_pca, filename = pca_file, dpi=300)
+
+
+# In[17]:
+
+
+# Plot - black
+g_pca = ggplot(all_data_df, aes(x='PC1', y='PC2')) + geom_point(aes(color='No. of experiments'), alpha=0.1) + facet_wrap('~Comparison') + labs(x = "PC 1", y = "PC 2", title = "PCA of experiment 1 vs multiple experiments") + theme(
+    plot_background=element_rect(fill='black'),
```

Comment:
The equivalent command in plotnine is ```theme_dark``` but this only sets the background to a dark grey instead of white and I was looking to make the backgrounds specifically black for use in presentations with black backgrounds

Reply:
ggdark has options for black background I think... 🤔 However, it may not be usable in python. Not a big deal anyway. 

You may want to save your dark theme preferences as a function in a script so that you can reuse it for other plotting tasks though. e.g. you could start a repo where you collect generally useful functions with input arg as general as possible (limited or absent hard-coded dependencies and parameters). Then you don't have to re-write this every time.